groups/bsl/bslstl/bslstl_monostate.t.cpp

```diff
@@ -0,0 +1,309 @@
+        //:   all relational operators as required by the standard.
+        //
+        // Plan:
+        //: 1 Apply each unary 'noexcept' operator to each relational
+        //:   operator, confirm that calculated boolean value matches the
+        //:   expected value.
+        //
+        // Testing:
+        //   CONCERN: Methods qualified 'noexcept' in standard are so.
+        // --------------------------------------------------------------------
+
+        if (verbose)
+            printf("\n'noexcept' SPECIFICATION"
+                   "\n========================\n");
+
+        bsl::monostate lhs;
+        bsl::monostate rhs;
+
+        ASSERT(BSLS_KEYWORD_NOEXCEPT_AVAILABLE ==
+               BSLS_KEYWORD_NOEXCEPT_OPERATOR(lhs == rhs));
+        ASSERT(BSLS_KEYWORD_NOEXCEPT_AVAILABLE ==
+               BSLS_KEYWORD_NOEXCEPT_OPERATOR(lhs != rhs));
+        ASSERT(BSLS_KEYWORD_NOEXCEPT_AVAILABLE ==
+               BSLS_KEYWORD_NOEXCEPT_OPERATOR(lhs < rhs));
+        ASSERT(BSLS_KEYWORD_NOEXCEPT_AVAILABLE ==
+               BSLS_KEYWORD_NOEXCEPT_OPERATOR(lhs > rhs));
+        ASSERT(BSLS_KEYWORD_NOEXCEPT_AVAILABLE ==
+               BSLS_KEYWORD_NOEXCEPT_OPERATOR(lhs <= rhs));
+        ASSERT(BSLS_KEYWORD_NOEXCEPT_AVAILABLE ==
+               BSLS_KEYWORD_NOEXCEPT_OPERATOR(lhs >= rhs));
+
+      } break;
+
+      case 3: {
+        // --------------------------------------------------------------------
+        // Relational operators
+        //
+        // Concerns:
+        //: 1 Any two 'monostate' objects compare equal
+        //: 2
+        //
+        // Plan:
+        //: 1 create two 'monostate' objects, and check that all comparison
+        //:   operators behave as expected. [C-1]
+        //
+        // Testing: operator==(monostate, monostate) operator!=(monostate,
+        // monostate) operator<(monostate, monostate) operator>(monostate,
+        // monostate) operator<=(monostate, monostate) operator>=(monostate,
+        // monostate)
+        // --------------------------------------------------------------------
+        if (verbose)
+            printf("\nRelational operators"
+                   "\n===================\n");
+        {
+            bsl::monostate lhs;
+            bsl::monostate rhs;
+
+            ASSERT((lhs == rhs));
+            ASSERT(!(lhs != rhs));
+            ASSERT(!(lhs < rhs));
+            ASSERT(!(lhs > rhs));
+            ASSERT((lhs <= rhs));
+            ASSERT((lhs >= rhs));
+        }
+
+      } break;
+      case 2: {
+        // --------------------------------------------------------------------
+        // Traits
+        //
+        // Concerns:
+        //: 1 'monostate' object is default constructible as required by the
+        //:   standard.
+        //: Plan:
+        //: 1 default construct an object of 'monostate' type.[ C-1]
+        //
+        // Testing:
+        //   CONCERN: defaulted default constructor is available
```

Comment:
Needs to match the string after the `[ 2]` at the top of the file.
```suggestion
        //   CONCERN: 'monostate' is default constructible
```